diff --git a/sysdeps/unix/sysv/linux/fchmodat.c b/sysdeps/unix/sysv/linux/fchmodat.c
index dad2e12..cf343a0 100644
--- a/sysdeps/unix/sysv/linux/fchmodat.c
+++ b/sysdeps/unix/sysv/linux/fchmodat.c
@@ -97,7 +97,7 @@ fchmodat (int fd, const char *file, mode_t mode, int flag)
     return INLINE_SYSCALL_CALL (fchmodat, fd, file, mode);
 
   int r = INLINE_SYSCALL_CALL (fchmodat2, fd, file, mode, flag);
-  if (r != 0 && errno == ENOSYS)
+  if (r != 0 && (errno == ENOSYS || errno == EPERM))
     return fchmodat_fallback (fd, file, mode, flag);
   return r;
 #endif
