diff --git a/test.c b/test.c
index 20b2775..8ab56cc 100644
--- a/test.c
+++ b/test.c
@@ -157,16 +157,8 @@ int main(int argc, char *argv[])
     assert_mktemp_path(TESTDIR "/temp", "", buf);
 
     test_spawn();
-    test_system();
     test_stat_with_null_path();
 
-    // Only run subprocess if no arguments are given
-    // as the subprocess will be called without argument
-    // otherwise we will have infinite recursion
-    if (argc == 1) {
-        test_subprocess();
-    }
-
     test_execv();
 
     /* If all goes well, this is never reached because test_execv() replaces
